### PR TITLE
#167511443: Setup Postgresql client.

### PR DESCRIPTION
**Pull request by fahdjamy**
#### Description

- add a Postgres client.

#### Type of change

Please select the relevant option

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Others (cosmetics, styling, improvements)
- [ ] This change requires a documentation update

#### How Has This Been Tested?

- [ ] Unit
- [ ] Integration
- [ ] End-to-end

#### How to Test

Give instructions on how reviewers can verify your work

#### Story-ID

[Delivers #167511443]

### CODE REVIEW

File: .sequelizerc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const path = require("path");
+
+module.exports = {
+  config: path.resolve("src", "db", "config", "config.js"), // can be also a .js file
```
Comment (MuhweziDeo) [inline .sequelizerc]: change this to `config.ts`

File: package.json [hunk 2]
```diff
@@ -26,7 +26,15 @@
...
     "@sentry/node": "^5.5.0",
-    "express": "^4.17.1"
+    "@types/sequelize": "^4.28.4",
+    "dotenv": "^8.0.0",
+    "express": "^4.17.1",
+    "pg": "^7.12.0",
+    "pg-hstore": "^2.3.3",
+    "reflect-metadata": "^0.1.13",
+    "sequelize": "^5.12.3",
+    "sequelize-cli-typescript": "^3.2.0-c",
```
Comment (MuhweziDeo) [inline package.json]: This should be a dev dependency

File: package.json [hunk 3]
```diff
@@ -26,7 +26,15 @@
...
+    "sequelize-typescript": "^1.0.0-beta.3"
```
Comment (MuhweziDeo) [inline package.json]: Dev dependency

File: src/db/migrations/20190805132033-create-user.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      updatedAt: {
+        allowNull: false,
+        type: Sequelize.DATE
+      }
+    });
+  },
+  down: (queryInterface, Sequelize) => {
+    return queryInterface.dropTable('Users');
+  }
+};
```
Comment (MuhweziDeo) [inline src/db/migrations/20190805132033-create-user.js]: EOL

File: src/db/models/user.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  const User = sequelize.define('User', {
+    firstName: DataTypes.STRING,
+    lastName: DataTypes.STRING,
+    email: DataTypes.STRING
+  }, {});
+  User.associate = function(models) {
+    // associations can be defined here
+  };
+  return User;
+};
```
Comment (MuhweziDeo) [inline src/db/models/user.js]: EOL

**Review by MuhweziDeo — CHANGES_REQUESTED** [review]
Nice Job

File: src/db/config/config.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
+const dotEnv = require('dotenv');
```
Comment (MuhweziDeo) [inline src/db/config/config.ts:1]: No need for this import here u can just call `require('dotenv).config()` or use es6 imports if u want to stick to importing it

File: src/db/models/index.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const Sequelize = require('sequelize');
+const basename = path.basename(__filename);
+const env = process.env.NODE_ENV || 'development';
+const config = require(__dirname + '/../config/config.js')[env];
```
Comment (MuhweziDeo) [inline src/db/models/index.ts]: maybe u missed this it should `config.ts`

File: src/db/models/index.ts [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+const db = {
+  sequelize: null,
+  Sequelize: null,
+};
+
+let sequelize;
+if (config.use_env_variable) {
```
Comment (MuhweziDeo) [inline src/db/models/index.ts:15]: Is this really necessary

Comment (MuhweziDeo) [reply src/db/models/index.ts:15]: the checking

**Review by MuhweziDeo — CHANGES_REQUESTED** [review]
Good Job

File: src/db/models/index.ts (hunk 8 above)
Comment (fahdjamy) [reply src/db/models/index.ts:15]: yes.

File: src/db/config/config.ts (hunk 6 above)
Comment (fahdjamy) [reply src/db/config/config.ts:1]: I get editor red squiggly lines in my editor that's why we gotta import everything. so this to me is flagged red `require('dotenv).config()`.

**Review by MuhweziDeo — APPROVED** [review]

**Pull request merged**